src/charm.py

```diff
@@ -216,6 +220,39 @@ def _get_istioctl(self) -> Istioctl:
             setting_overrides=setting_overrides,
         )
 
+    # This is a hacky way to get istio CNI to use REDIRECTION instead of TPROXY
+    # TODO: Remove this once we upgrade to istio 1.24 as REDIRECTION will be used by default
+    # Istioctl doesn't yet support adding env vars directly to the CNI component
+    def _patch_istio_cni_daemonset(self):
```

Comment:
What about moving this patch [`_reconcile_control_plane()`](https://github.com/canonical/istio-k8s-operator/blob/26ddfbff3e206439fa77a56b533e3fa4cd5b9406/src/charm.py#L123)?  That function:
1. generates the manifests
2. creates resources from them
3. applies them via the `krm`

between (2) and (3), you can modify the daemonset (essentially like how you do here) to apply the patch.  That way it isn't getting modified after being put into the cluster